Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp

```diff
@@ -87,10 +117,11 @@ void ABenchmarkGymGameMode::Tick(float DeltaSeconds)
 	{
 		if (NPCSToSpawn > 0)
 		{
-			int32 Cluster = (--NPCSToSpawn) % NumPlayerClusters;
+			int32 NPCIndex = --NPCSToSpawn;
+			int32 Cluster = (NPCIndex) % NumPlayerClusters;
 			int32 SpawnPointIndex = Cluster * PlayerDensity;
 			const AActor* SpawnPoint = SpawnPoints[SpawnPointIndex];
-			SpawnNPC(SpawnPoint->GetActorLocation());
+			SpawnNPC(SpawnPoint->GetActorLocation(), NPCRunPoints[NPCIndex % NPCRunPoints.Num()]);
```

Comment:
Curious - why do we do this in tick, and not BeginPlay ?

Reply:
I had a theory that there were some flakes caused by a spike on the server but I don't think this was the cause. We could move it back but I suppose spawning 300+ actors with nav queries is still quite heavy

Reply:
Ah okay, yeah could be cleaned up, but not necessary.